packages/flutter_tools/lib/src/test/flutter_platform.dart

```diff
@@ -240,57 +239,68 @@ class _Compiler {
 
     printTrace('Listening to compiler controller...');
     compilerController.stream.listen((_CompilationRequest request) async {
-      final bool isEmpty = compilationQueue.isEmpty;
       compilationQueue.add(request);
       // Only trigger processing if queue was empty - i.e. no other requests
       // are currently being processed. This effectively enforces "one
       // compilation request at a time".
-      if (isEmpty) {
-        while (compilationQueue.isNotEmpty) {
-          final _CompilationRequest request = compilationQueue.first;
-          printTrace('Compiling ${request.path}');
-          final Stopwatch compilerTime = Stopwatch()..start();
-          bool firstCompile = false;
-          if (compiler == null) {
-            compiler = createCompiler();
-            firstCompile = true;
-          }
-          suppressOutput = false;
-          final CompilerOutput compilerOutput = await handleTimeout<CompilerOutput>(
-              compiler.recompile(
-                  request.path,
-                  <String>[request.path],
-                  outputPath: outputDill.path),
-              request.path);
-          final String outputPath = compilerOutput?.outputFilename;
-
-          // In case compiler didn't produce output or reported compilation
-          // errors, pass [null] upwards to the consumer and shutdown the
-          // compiler to avoid reusing compiler that might have gotten into
-          // a weird state.
-          if (outputPath == null || compilerOutput.errorCount > 0) {
-            request.result.complete(null);
-            await _shutdown();
-          } else {
-            final File outputFile = fs.file(outputPath);
-            final File kernelReadyToRun = await outputFile.copy('${request.path}.dill');
-            final File testCache = fs.file(testFilePath);
-            if (firstCompile || !testCache.existsSync() || (testCache.lengthSync() < outputFile.lengthSync())) {
-              // The idea is to keep the cache file up-to-date and include as
-              // much as possible in an effort to re-use as many packages as
-              // possible.
-              ensureDirectoryExists(testFilePath);
-              await outputFile.copy(testFilePath);
-            }
-            request.result.complete(kernelReadyToRun.path);
-            compiler.accept();
-            compiler.reset();
+      ResidentCompiler compiler;
+      String outputDill;
+      bool firstCompile = false;
+      if (idleCompilers.isEmpty && compilers < concurrency ) {
+        // Create compiler
+        compiler = createCompiler();
+        firstCompile = true;
+        compilers++;
+        outputDill = outputDillDirectory.childFile('output_$compilers.dill').path;
+        printTrace('Compiler will use the following file as its incremental dill file: $outputDill');
```

Comment:
We are going to see multiples of these trace messages depending on the number of compilers being used, maybe the message should also say which compiler number it is referring to
'Compiler($compilers) will use ...'